SIGs/GRC/publications/Confidential Payload Upgrade Governance.md

```diff
@@ -0,0 +1,46 @@
+# ![CCC GRC logo](./images/ccc_grc_logo.png) Confidential Payload Upgrade Governance
+
+# Context
+
+Distributed applications, both in the cloud and IoT, are frequently deployed as a large number of nearly-identical instances. Furthermore, such applications are likely to be less tolerant of downtime. When such applications get updated, the update is usually rolled out gradually, and in some cases when a problem is found with the upgrade, a rollback may be required. During the upgrade, both old and new versions of the application are likely to coexist. For the successful continuous operation of the overall system, it needs to properly handle intermediate states where a subset of the application instances is upgraded, while the complementary subset of application instances is still awaiting upgrades.
+
+# Problem
+
+Ensuring that the overall system – starting with Verifier and ending with Relying Parties – continues to operate without incurring undue downtime during the upgrade/rollback period requires careful orchestration.
+
+# Forces
+
+Critical distributed applications often have high availability requirements. In addition, Control Specifications or Control Objectives may dictate that for certain classes of problems (e.g., high-impact vulnerabilities), all systems must be updated within a given time interval, sometimes an aggressive one.
+
+# Solution
+
+The terms MUST/SHOULD/MAY etc. below are used in accordance with **\[2\]**. Every SHOULD recommendation is explained separately in the “SHOULD vs. MUST Clarifications” section towards the end of this document.
+
+Integrate the upgrade with the software deployment stack (e.g., the CI/CD pipeline).  
+Once a new Confidential Payload[^1] version becomes available, the Verifier policies[^2] SHOULD **\[a\]** allow both old and new versions of the Payload to be treated as valid. If the rollout of the new Payload version is successful, the old Payload version is then removed from the Verifier policies. Conversely, if the rollout experiences difficulties, a rollback is initiated and the new Payload version is removed from the Verifier policies leaving only the old Payload version as valid. This is illustrated in the diagram below.
```

Comment:
Do you want to embrace the model where old/rolled back workload versions are not removed, but are instead detected? This may assist in workload management. Instead of removing the reference values, endorsed metadata is provided to the verifier allowing it to detect the use of deprecated versions (as distinct from unrecognised versions)? The verifier can then add attestation result entries that reflect this condition.

Reply:
What would be the practical difference from the point of view of the relying party? The relying party cares that its authorization policy is extra stable, e.g. "grant access to Payroll application". What constitutes the Payroll application is determined by the Verifier. So whether Verifier looks at Reference Values and thinks that they are unrecognized or deprecated, the result ought to be the same -- failed attestation. Do you not agree?

Reply:
@SimonFrost-Arm is this resolved?

Reply:
Discussion in TAC 5/15. This level of detail may be lower than the Patterns level defines. Different means to similar ends. We resolving the comment without change to the text today, but please add an issue to reopen this topic if necessary.